inductiva/_benchmarks/benchmark.py

```diff
@@ -133,38 +141,78 @@ def wait(self) -> Self:
             task.wait(download_std_on_completion=False)
         return self
 
-    def to_dict(self):
+    def export(
+        self,
+        format: ExportFormat = ExportFormat.JSON,
+        filename: Optional[str] = None,
+    ):
+        """
+        Exports the benchmark results in the specified format.
+
+        Args:
+            format (str): The format to export the results in. Supported 
+                formats include: 'json'.
+            filename (Optional[str]): The name of the output file to save the
+                exported results. Defaults to the benchmark's name if not
+                provided.
         """
-        Compiles results from all completed tasks into a dictionary.
+        data = self.gather_data(verbose=False)
+        filename = filename or f"{self.name}.{format.value}"
+        match format:
+            case ExportFormat.JSON:
+                with open(filename, mode="w", encoding="utf-8") as file:
+                    json_content = json.dumps(obj=data, indent=4)
+                    file.write(json_content)
+            case ExportFormat.CSV:
+                with open(filename, mode="w", encoding="utf-8") as file:
+                    csv_content = [{
+                        "machine_type": machine_type,
+                        **task
+                    } for machine_type, tasks in data.items() for task in tasks]
+                    fieldnames = csv_content[0].keys()
+                    writer = csv.DictWriter(file, fieldnames=fieldnames)
+                    writer.writeheader()
+                    writer.writerows(csv_content)
+            case _:
+                raise ValueError(f"Unsupported export format: {format}")
+
+    def gather_data(self, verbose: bool = False) -> dict:
+        """
+        Gathers information, including metrics and metadata, from all tasks.
+
+        Args:
+            verbose (bool): A boolean indicating whether to include additional
```

Comment:
Usually verbose is used to enable/disable prints. In this case verbose is controlling the information returned. We should pick another name for this. Maybe `simulation_arguments` or `gather_simulation_arguments`